client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -246,6 +283,88 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
         batchHandleChangePartitionRequestInterval,
         TimeUnit.MILLISECONDS)
     }
+
+    batchHandleCommitPartitionSchedulerThread.foreach {
+      _.scheduleAtFixedRate(
+        new Runnable {
+          override def run(): Unit = {
+            logWarning("Batch commit hard split partition")
+            commitPartitionRequests.asScala.foreach { case (shuffleId, requests) =>
+              batchHandleCommitPartitionExecutors.submit {
+                new Runnable {
+                  override def run(): Unit = {
+                    val shuffleCommittedInfo = committedPartitionInfo.get(shuffleId)
```

Comment:
I think we can extract a method to handle both stage_end and early commit scenarios. Because no matter what kind of scenario, our processing process would be similar, and we all need to care about whether there is datalost, and we can record committed file groups/filecount, etc.  for example  `` private def handleCommitFiles(applicationId: String,
      shuffleId: Int,
      allocatedWorkers: util.Map[WorkerInfo, PartitionLocationInfo],
      partitionUniqueIdOpt: Option[util.HashSet[String]] = None)``

Reply:
I know, but I prefer to do this refactor next, I don't want one pr to be too complex, refactor this based on this pr can be more clear. WDYT @waitinfuture